### DISCUSSION TITLE
Adding options in create() - Ex - Default Comparison - shallow.

### DISCUSSION THREAD

**Discussion by astrarudra** [pmndrs/zustand#2490] [Ideas] [answered]
Hi, 

So here is the idea - 
Currently, we need to import and use useShallow everywhere  -  

```
const { nuts, honey } = useBearStore(
  useShallow((state) => ({ nuts: state.nuts, honey: state.honey })),
)
```

Instead why not allow an option for the user to set a  default comparison in create. 
And then override if nessesary.

```
const useBearStore = create((set) => ({
  bears: 0,
  increasePopulation: () => set((state) => ({ bears: state.bears + 1 })),
  removeAllBears: () => set({ bears: 0 }),
}), {
  equality: shallow
})
```

Idea is from Zundo. It takes in equality: shallow on initiation.
Also, having options wil help in extensibility - we can add more options in future.

**Comment by dai-shi** [comment, marked as answer]
That's exactly what `createWithEqualityFn` does. Maybe we should update the docs. cc @dbritto-dev

**Reply by dbritto-dev** [reply]
@dai-shi yeah, totally

**Reply by astrarudra** [reply]
Oh - any example of this createWithEqualityFn ? 
Thanks - this is great!

**Reply by dbritto-dev** [reply]
@astrarudra sure

```ts
import { createWithEqualityFn, shallow  } from 'zustand/traditional'

const store = createWithEqualityFn((...args) => ({ ...  }), shallow)
```

**Reply by dai-shi** [reply]
from 'zustand/traditional'

**Reply by astrarudra** [reply]
shallow is not in 'zustand/traditional'

**Reply by dai-shi** [reply]
```js
import { createWithEqualityFn } from 'zustand/traditional'
import { shallow } from 'zustand/shallow'
```
